cpp/src/reductions/scan/scan_inclusive.cu

```diff
@@ -142,10 +149,23 @@ struct scan_functor<Op, cudf::string_view> {
       result.begin(),
       min_max_scan_operator<cudf::string_view, Op>{*d_input, input_view.has_nulls()});
 
+    if (input_view.has_nulls()) {
+      // fill the null entries with out-of-bounds value so gather records them as null
+      // this prevents un-sanitized null entries in the output
+      auto null_itr = detail::make_counting_transform_iterator(0, null_iterator{mask});
+      auto oob_val  = thrust::constant_iterator<size_type>(input_view.size());
+      thrust::scatter_if(rmm::exec_policy(stream),
+                         oob_val,
+                         oob_val + input_view.size(),
+                         thrust::counting_iterator<size_type>(0),
+                         null_itr,
+                         result.data());
```

Comment:
`result` here is not the function result. It is just an intermediate gather map. So we should better rename it into `result_map` or `gather_map` or something similar.